### ISSUE TITLE
User ID is ignored in partial import

### ISSUE THREAD

**Issue by mvalensky** [keycloak/keycloak#19257]
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the latest release

### Area

import-export

### Describe the bug

During partial import, the user ID is ignored and generated.

### Version

21.0.1

### Expected behavior

If the user ID is specified in partial import, it is taken into account and the user with the ID is created.

### Actual behavior

The user ID is always ignored and generated.

### How to Reproduce?

Perform partial import of the following file in the Admin console:
``` json
{
  "users": [
    {
      "id": "8281cec3-5b48-46eb-a41d-72c15ec3f9e0",
      "username": "test-import",
      "enabled": true,
      "credentials": [
        {
          "type": "password",
          "value": "password"
        }
      ],
      "realmRoles": ["user"],
      "clientRoles": {
        "account": ["view-profile", "manage-account"]
      }
    }
  ]
}
```

**Comment by mvalensky** [comment]
Sure, I don't exploit the migration for backups, but I import local users from old IDP and need to keep user IDs. Otherwise, the references to users are broken in all my apps.

**Comment by fcovatti** [comment]
@ssilvert I am migrating from an old version of KC and faced this same issue, so I kind of either disagree that the partial import is not be used as a backup or I am doing the transition of versions wrongly. If this is not something fixable I would document that his happens :)

**Comment by fcovatti** [comment]
Not sure if this is related with the user ID being diferently, but we have verified also that the Credentials were not imported properly on version 21.0.1, meaning all previous user passwords do not work anymore. Searching here I could not find such issue in the history.

**Comment by hmlnarik** [comment]
This seems a documentation issue.
Summarizing:
- Partial export/import is not meant to be a backup replacement
- User IDs are not preserved, which is addressed in #20419
- Admin console export / import of a user is not expected to retain the credentials. See https://github.com/keycloak/keycloak/blob/d543ba5b56db9d301dd000fb462370d98e77e601/docs/documentation/server_admin/topics/export-import.adoc?plain=1#L115
